app/src/main/java/org/mozilla/fenix/perf/DurationContainer.kt

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.os.SystemClock
+
+private const val MSG_ASSERT_UNSET = "this method should only be called once"
+
+typealias MeasuredFunction<R> = () -> R
+
+/**
+ * A class to measure the duration between two points in time in a less error-prone way.
+ *
+ * The error we're mitigating is loss of data integrity. This is a concern because
+ * telemetry implementations are often not unit tested or manually retested by
+ * programmers or users so they're likely to fail.
+ *
+ * This class helps ensure data integrity by making guarantees about how it's supposed
+ * to be used since usage outside of these expectations is often indicative of faulty
+ * assumptions or programming errors. We crash in these cases, rather than sending an
+ * error to the server, to reduce the feedback time between change in implementation and
+ * new errors, to increase the likelihood the errors are noticed and fixed rather than
+ * being used for bad data analyses, and to reduce the amount of bad data hitting the server
+ * since it would complicate data analysis and make it more error-prone. It's unfortunate to
+ * crash crash over data collection concerns but the trade-offs seem worth it.
+ *
+ * One large downside to these guarantees is that this class will become harder to use.
```

Comment:
It's purely theoretical: with complex classes sometimes devs write their own simple version rather than figuring out how to use the complex one. Then again, those classes are usually much larger than this one.

Reply:
Removed this class so nevermind. :)